cmd/acr/purge.go

```diff
@@ -568,43 +569,45 @@ func dryRunPurge(ctx context.Context, acrClient api.AcrCLIClientInterface, login
 			}
 			return -1, -1, err
 		}
-		// This will act as a set if a key is present then it should not be deleted because it is referenced by a multiarch manifest
-		// that will not be deleted
-		doNotDelete := map[string]bool{}
+		// This set will do the check: if a key is present, then the corresponding manifest
+		// should not be deleted because it is referenced by a multiarch manifest
+		// or otherwise references a subject
+		doNotDelete := set.New[string]()
 		candidatesToDelete := []acr.ManifestAttributesBase{}
-		// Iterate over all manifests to discover multiarchitecture manifests
 		for resultManifests != nil && resultManifests.ManifestsAttributes != nil {
 			manifests := *resultManifests.ManifestsAttributes
 			for _, manifest := range manifests {
-				// If the manifest is manifest list and would not get deleted then mark it's dependant manifests as not deletable.
-				if (*manifest.MediaType == manifestListContentType || *manifest.MediaType == manifestIndexContentType) && (*countMap)[*manifest.Digest] != deletedTags[*manifest.Digest] {
-					var manifestListBytes []byte
-					manifestListBytes, err = acrClient.GetManifest(ctx, repoName, *manifest.Digest)
+				if countMap[*manifest.Digest] != deletedTags[*manifest.Digest] {
+					// If the number of tags associated with the manifest is not equal to
+					// the number of tags to be deleted associated with the said manifest,
+					// we will need to marked the dependent manifests (if any) to not be deleted
+					err = getDoNotDeleteListFromMultiArchManifestWithTag(ctx, manifest, doNotDelete, acrClient, repoName)
 					if err != nil {
 						return -1, -1, err
 					}
-					var manifestList multiArchManifest
-					err = json.Unmarshal(manifestListBytes, &manifestList)
+				} else {
+					// If the number of tags associated with the manifest is equal to
```

Comment:
nit: We can just call out that this indicates that this equivalent to that the manifest has no tag.